src/tracing/mod.rs

```diff
@@ -253,10 +324,18 @@ impl<P: EthereumProvider + Send + Sync + Clone> Tracer<P> {
                         TracingOptions::Parity(tracing_config) => {
                             Self::trace_parity(env, &mut db, tx, *tracing_config)?
                         }
+                        TracingOptions::GethCall(_) => {
+                            return Err(EthApiError::Transaction(TransactionError::Tracing(
+                                eyre!("`TracingOptions::GethCall` is not supported in `trace_transactions` context")
+                                    .into(),
+                            )))
+                        }
                     }
                 };
 
-            traces.extend(convert_result(res).unwrap_or_default());
+            if let Some(result) = convert_result(&res) {
+                traces.extend(result.iter().cloned());
```

Comment:
I think it's better to use `clone` and `append` here

https://github.com/rust-lang/rust-clippy/issues/4321#issuecomment-929110184

Reply:
fixed